### PR TITLE
Introduce TimeOut waiter

### PR DESCRIPTION
**Pull request by tyx**
Let me know if any change should be done to be merged

### CODE REVIEW

File: src/Tolerance/Waiter/TimeOut.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+     */
+    private $delegateWaiter;
+
+    /**
+     * @var integer
+     */
+    private $timeOut;
+
+    /**
+     * var float
```
Comment (sroze) [inline src/Tolerance/Waiter/TimeOut.php]: Can you add a `@` before `var` ?

File: src/Tolerance/Waiter/TimeOut.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+     */
+    private $secondsEllapsed;
+
+    public function __construct(Waiter $delegateWaiter, $timeOut)
```
Comment (sroze) [inline src/Tolerance/Waiter/TimeOut.php]: Can you add a PhpDoc on the constructor?

File: src/Tolerance/Waiter/TimeOut.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        $this->delegateWaiter = $delegateWaiter;
+        $this->timeOut = $timeOut;
+        $this->secondsEllapsed = 0;
```
Comment (sroze) [inline src/Tolerance/Waiter/TimeOut.php]: `0.0` if your `secondsEllapsed` property is actually float.

Comment (tyx) [inline src/Tolerance/Waiter/TimeOut.php]: But we definitively should consider to switch the `$seconds` in `Waiter::wait` to `float` also. I currently use `0.5` in my own app. And this is why I managed `secondsEllapsed` in float

**Comment by tyx** [comment]
changes done

**Comment by sroze** [comment]
Thank you @tyx!

**Pull request merged**